localstack-core/localstack/aws/protocol/serializer.py

```diff
@@ -614,6 +657,30 @@ def _add_md5_header(self, response: Response):
     def _get_error_message(self, error: Exception) -> str | None:
         return str(error) if error is not None and str(error) != "None" else None
 
+    def _get_error_status_code(
+        self, error: ServiceException, headers: Headers, service_model: ServiceModel
```

Comment:
nit: will update type hint of `headers` as the caller `serialize_error_to_response` defines `headers: dict | Headers | None`